spec/enumerables_spec.rb

```diff
@@ -0,0 +1,174 @@
+# frozen_string_literal: true
+
+require './lib/enumerable.rb'
+
+describe Enumerable do
+  let(:arr) { [1, 2, 7, 8, 5] }
+
+  describe '#my_each should return' do
+    it 'an array holding all elements of the given array' do
+      expect(arr.my_each).to eql([1, 2, 7, 8, 5])
+    end
+
+    it 'an empty array if an empty array is given' do
+      expect([].my_each).to eql([])
+    end
+
+    it 'an array of the yield for each element if given a block' do
+      result = arr.my_each { |x| x + 2 }
+      expect(result).to eql([3, 4, 9, 10, 7])
+    end
+  end
+
+  describe '#my_each_with_index should return' do
+    it 'the same array if no block is given' do
+      result = arr.my_each_with_index
+      expect(result).to eql([1, 2, 7, 8, 5])
+    end
+
+    it 'an empty array if an empty array is given' do
+      expect([].my_each_with_index).to eql([])
+    end
+
+    it 'the yield for each (element, index) pair in the given array if block is given' do
+      holder = []
+      arr.my_each_with_index do |i, idx|
+        holder << i + idx
+      end
+      expect(holder).to eql([1, 3, 9, 11, 9])
+    end
+  end
+
+  describe '#my_select should return' do
+    it 'array of values that return true for the block' do
+      result = arr.my_select { |i| i.even? }
+      expect(result).to eql([2, 8])
+    end
+
+    it 'array of values that return true for the block' do
+      result = %w[a b D].my_select { |i| i.match?(/^[A-Z]{1}$/) }
+      expect(result).to eql(['D'])
+    end
+
+    it 'empty array if given an empty array' do
+      result = [].my_select { |i| i.match?(/^[A-Z]{1}$/) }
+      expect(result).to eql([])
+    end
+
+    it 'the same array if no block is given' do
+      expect(arr.my_select).to eql([1, 2, 7, 8, 5])
```

Comment:
*Suggestion*
Maybe you can try a test that runs against original select
SOmething like this would be nice 👍 
`expect(arr.my_select).to eql(arr.select)`

Reply:
Yeah, this is cool. :grin: